package/samplers/cmamae/sampler.py

```diff
@@ -179,18 +180,27 @@ def after_trial(
         self._validate_param_names(trial.params.keys())
 
         # Store the trial result.
-        self._values_to_tell.append(values)
+        direction0 = study.directions[0]
+        minimize_in_optuna = study.StudyDirection.MINIMIZE == direction0
+        assert values is not None, "MyPy redefinition."
```

Comment:
This is also necessary unfortunately because `values` might not exist when we consider `FAILED` trials.
However, for this version, we assume that we will not get failed trials.